### PR TITLE
Build warnings Travis

### PR DESCRIPTION
**Pull request by EShamaev**
These warnings are very annoying when looking at Travis output.

### CODE REVIEW

**Comment by khancyr** [comment]
LGTM

File: libraries/AP_Baro/examples/ICM20789/ICM20789.cpp [hunk 1]
```diff
@@ -35,6 +35,9 @@ static AP_HAL::OwnPtr<AP_HAL::Device> dev;
 #define CMD_SOFT_RESET     0x805D
 #define CMD_READ_ID        0xEFC8
 
+void setup(void);
+void loop(void);
```
Comment (OXINARF) [inline libraries/AP_Baro/examples/ICM20789/ICM20789.cpp]: Where are you seeing a warning with these two? The only warnings I saw with this example were:

```
../../libraries/AP_Baro/examples/ICM20789/ICM20789.cpp:41:13: warning: ‘void spi_init()’ defined but not used [-Wunused-function]
 static void spi_init()
             ^
../../libraries/AP_Baro/examples/ICM20789/ICM20789.cpp:219:13: warning: ‘void i2c_init()’ defined but not used [-Wunused-function]
 static void i2c_init(void)
```

Which will continue to happen.

Comment (EShamaev) [reply libraries/AP_Baro/examples/ICM20789/ICM20789.cpp]: image

**Comment by tridge** [comment]
this misses two uses of ifdef for HAL_RCINPUT_WITH_AP_RADIO

**Comment by khancyr** [comment]
push the missing occurances

File: libraries/AP_Baro/examples/ICM20789/ICM20789.cpp (hunk 1 above)
Comment (EShamaev) [reply libraries/AP_Baro/examples/ICM20789/ICM20789.cpp]: Warnings are from Travis px4.

**Comment by magicrub** [comment]
needs a rebase, otherwise LGTM

**Comment by khancyr** [comment]
force push rebase . Let's merge !

**Comment by OXINARF** [comment]
I have an alternative branch at https://github.com/OXINARF/ardupilot/commits/review/7896

Changes are:
- dropping last commit from @khancyr; it doesn't make sense to me to check for HAL_RCINPUT_WITH_AP_RADIO existence twice; more importantly though, the definitions in px4.h aren't really dependent on that macro, but on the exact board; because there is no macro for skyviper-v2450-px4, I've moved those definitions to the board config in waf. I'll run this one by @tridge and if he agrees, I'll merge it
- the AP_Baro commit also protects spi_init and i2c_init that gave unused warnings in some builds

**Comment by khancyr** [comment]
hummm, the last commit should only change the #ifdef to #if . The problem is that on gcc 7, if it doesn't know the definition it defines it to 0. Otherwise it is good

**Comment by khancyr** [comment]
force push with @OXINARF modifications !

**Comment by khancyr** [comment]
rebased on master

**Comment by rmackay9** [comment]
Looks like everyone is happy so merging.

**Pull request merged**